tt_metal/include/compute_kernel_api/state_tracker.h

```diff
@@ -0,0 +1,307 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "reconfig_data_format.h"
+#include "tt_metal/include/compute_kernel_api/common_globals.h"
+
+#define RECONFIG_NOTHING_CHANGED 0x00
+#define RECONFIG_CHANGED_SRCA 0x01
+#define RECONFIG_CHANGED_SRCB 0x02
+#define RECONFIG_CHANGED_PACK 0x04
+
+#ifdef TT_METAL_STATE_TRACKER_TESTING_ENABLED
+#define TEST_RECONFIG_CALLS(expected_calls) reconfigure_tester.testReconfigureCalls(expected_calls)
+#define SET_CALLED_RECONFIG(call_bit) reconfigure_tester.reconfigCalls |= (call_bit)
+#else
+#define TEST_RECONFIG_CALLS(expected_calls) false
+#define SET_CALLED_RECONFIG(call_bit)
+#endif
+
+namespace ckernel {
+
+/*
+ * State Tracking Overview:
+ *
+ * The state_configure functions are organized into three overloads based on operand count:
+ *
+ * 1) Single operand: state_configure(cb)
+ *    - Handles operations that only need one input buffer
+ *    - Reconfigures either srcA or pack registers depending on operation type
+ *
+ * 2) Dual operand: state_configure(cb_a, cb_b)
+ *    - Handles operations with two buffers
+ *    - Reconfigures srcA and srcB registers, or srcA and pack
+ *
+ * 3) Three operand: state_configure(cb_a, cb_b, cb_out)
+ *    - Handles full pipeline operations
+ *    - Reconfigures all three registers: srcA, srcB, and pack
+ *
+ * Each operation type uses compile-time dispatch (if constexpr) to call the appropriate
+ * reconfiguration helper, minimizing runtime overhead while maintaining flexibility.
+ */
+// Forward declarations to avoid circular dependency with pack.h
+ALWI void pack_reconfig_data_format(uint32_t new_cb_id);
+ALWI void pack_reconfig_data_format(uint32_t old_cb_id, uint32_t new_cb_id);
+
+/**
+ * These enum values are used with state_configure<Operation::TYPE>() to determine
+ * which hardware registers need reconfiguration.
+ */
+namespace Operation {
+enum Type : uint8_t {
```

Comment:
```suggestion
enum class OperationType : uint8_t {
```
This way you get your scope without a need for a nested namespace.